gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/javaversions/BaselineJavaVersion.java

```diff
@@ -198,6 +199,13 @@ public void execute(Test test) {
                 test.getJvmArgumentProviders().add(new EnablePreviewArgumentProvider(runtime));
             }
         });
+
+        project.getTasks().withType(Checkstyle.class).configureEach(new Action<Checkstyle>() {
+            @Override
+            public void execute(Checkstyle checkstyle) {
+                checkstyle.getJavaLauncher().set(javaToolchain.flatMap(BaselineJavaToolchain::javaLauncher));
+            }
+        });
```

Comment:
Mind adding a test that this resolves?

Reply:
I'm not entirely sure what this means. What sort of test would you like me to write? I'm not really sure how to write the Checkstyle equivalent of the Java one.